torch/utils/_foreach_utils.py

```diff
@@ -5,6 +5,59 @@
 from torch import Tensor
 from torch.autograd.grad_mode import no_grad
 
+__all__ = ["SupportForeachDevices"]
+
+class SupportForeachDevices(object):
+    r"""
+    A class that manages the default device type for foreach support,
+    default is List['cuda'].
+    """
+    _default_device_types = ["cuda"]
```

Comment:
I appreciate the class wrapping a list, but what's the advantage of having a class instead of just having this list be explicitly by itself? Could you go into detail about how the use case with a custom device?